### ISSUE TITLE
Unify fields with camelcase or snakecase

### ISSUE THREAD

**Issue by abdonrd** [strapi/strapi#8472]
### Please describe your feature request

- [ ] **I have created my request on the Product Board before I submitted this issue**
- [x] **I have looked at all the other requests on the Product Board before I submitted this issue**

### Summary

Unify the fields because some use camelcase and others snakecase:

`createdAt`
`updatedAt`

vs

`published_at`
`created_by`
`updated_by`

### Why is it needed?

For consistency.

### Suggested solution(s)

All with camelcase or all with snakecase.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request.

**Comment by derrickmehaffy** [comment]
I spoke with @alexandrebodin about this a little while ago on another issue (can't find it) but this change will be made when we transition into v4 as it's a massive breaking change for existing applications and requires a lot of migration.

**Comment by jpizzle34** [comment]
I would prefer snakecase as I tend to use postgresql the most and by default postgres will lowercase column names unless you surround them with double quotes (which itself leads to other issues when changing field names)

```
error error: select "steering_wheel_shapes".* from "steering_wheel_shapes" where LOWER(CAST("steering_wheel_shapes"."slug" AS VARCHAR)) LIKE LOWER($1) - column steering_wheel_shapes.slug does not exist
```

This occured after I migrated from sqlite to postgresql using pgloader, then changed a field name from guid to slug. I had to enable quote identifiers in pgloader to preserve the camelCase of strapi built-in fields (e.g. strapi.administrator.isActive). I think if you go with snakecase you will encounter less problems like this.

Looking forward to v4! Thank for your work guys!
